pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,830 @@
+from pyanaconda.modules.common.util import is_module_available
+from pyanaconda.modules.common.task import sync_run_task
+
+from pyanaconda.ui.gui.spokes import NormalSpoke
+from pyanaconda.ui.gui.spokes.lib.subscription import fill_combobox
+from pyanaconda.ui.categories.system import SystemCategory
+from pyanaconda.ui.communication import hubQ
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SubscriptionSpoke"]
+
+
+# the integers correspond to the order of options
+# in the authentication mode combo box
+class AuthenticationMethod(IntEnum):
+    USERNAME_PASSWORD = 0
+    ORG_KEY = 1
+
+
+class SubscriptionSpoke(NormalSpoke):
+    """Subscription spoke provides the Connect to Red Hat screen."""
+    builderObjects = ["subscription_window"]
+
+    mainWidgetName = "subscription_window"
+    uiFile = "spokes/subscription.glade"
+    help_id = "SubscriptionSpoke"
+
+    category = SystemCategory
+
+    icon = "application-certificate-symbolic"
+    title = CN_("GUI|Spoke", "_Connect to Red Hat")
+
+    # main notebook pages
+    REGISTRATION_PAGE = 0
+    SUBSCRIPTION_STATUS_PAGE = 1
+
+    @classmethod
+    def should_run(cls, environment, data):
+        """The Subscription spoke should run only if the Subscription module is available."""
+        return is_module_available(SUBSCRIPTION)
+
+    def __init__(self, *args):
+        super().__init__()
+
+        # connect to the Subscription DBus module API
+        self._subscription_module = SUBSCRIPTION.get_proxy()
+
+        # connect to the Network DBus module API
+        self._network_module = NETWORK.get_proxy()
+
+        # get initial data from the Subscription module
+        self._subscription_request = self._get_subscription_request()
+        self._system_purpose_data = self._get_system_purpose_data()
+        # Keep a copy of system purpose data that has been last applied to
+        # the installation environment.
+        # That way we can check if the main copy of the system purposed data
+        # changed since it was applied (for example due to user input)
+        # and needs to be reapplied.
+        # By default this variable is None and will only be set to a
+        # SystemPurposeData instance when first system purpose data is
+        # applied to the installation environment.
+        self._last_applied_system_purpose_data = None
+
+        self._authentication_method = AuthenticationMethod.USERNAME_PASSWORD
+
+        self._registration_error = ""
+        self._registration_phase = None
+        self._registration_controls_enabled = True
+
+        # Red Hat Insights should be enabled by default for non-kickstart installs.
+        #
+        # For kickstart installations we will use the value from the module, which
+        # False by default & can be set to True via the rhsm kickstart command.
+        if not flags.automatedInstall:
+            self._subscription_module.InsightsEnabled = True
```

Comment:
I think that you need to call `SetInsightsEnabled`.

Reply:
Indeed! Fixed. :)